DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h

```diff
@@ -0,0 +1,91 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_FDD_DIGIT_H
+#define ALICEO2_FDD_DIGIT_H
+
+#include "CommonDataFormat/InteractionRecord.h"
+#include "CommonDataFormat/TimeStamp.h"
+#include <iosfwd>
+#include "Rtypes.h"
+
+namespace o2
+{
+namespace fdd
+{
+
+struct ChannelData {
+  Int_t mPMNumber;    // PhotoMultiplier number (0 to 16)
+  Float_t mTime;      // Time of Flight
+  Float_t mWidth;     // Width of the time distribution
+  Short_t mChargeADC; // ADC sample as present in raw data
+  ClassDefNV(ChannelData, 1);
+};
+
+using DigitBase = o2::dataformats::TimeStamp<double>;
+class Digit : public DigitBase
+{
+ public:
+  Digit() = default;
+
+  Digit(std::vector<ChannelData> channelData, Double_t time, uint16_t bc, uint32_t orbit, std::vector<Bool_t> triggers)
```

Comment:
Do you really need to make a digit using existing outer containers? (Could the digit not manage it's own containers?)
I would in any case still prefer to have `const&` here.

Reply:
Its very handy like that... Where exactly do you want the const& , not clear to me?

Reply:
`Digit(std::vector<ChannelData> const& channelData,...)`. Without the reference, the vector will be copied into a tmp container when you call the function (as a general rule always pass complicated objects by reference and not by value).

Reply:
Ok, thanks for explanation. Done now.